### PR TITLE
Add component maturity page to guidelines

### LINKED ISSUE #372: Create component maturity level documentation
Component maturity documentation will help folks know how mature each component of the CMS Design System is. This will also allow for the CMS Design System to release components for teams to help test and validate before making sure they are ready to go.

### PR DESCRIPTION
**Pull request by line47**
Add component maturity info page to the docs
This closes #372 

### Screenshot
Screen Shot 2019-03-20 at 4 22 46 PM

### CODE REVIEW

File: packages/docs/src/pages/guidelines/component-maturity.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+## Work in progress
+- **Implement with caution**
+- Documentation and guidance is complete
+- Testing and validating is in progress
+- Could be buggy or have accessibility issues
+
+## Ready
+- **Ready to implement**
+- Documentation and guidance is complete
+- Testing and validating is complete
```
Comment (1Copenut) [inline packages/docs/src/pages/guidelines/component-maturity.md]: Let's change validating to validation on all three stops.

As a post-PR item, I think we could link these out to detailed explanations of what testing and validation look like. Same for accessibility.

Comment (line47) [reply packages/docs/src/pages/guidelines/component-maturity.md]: I think so too, what it means for each of those steps to be "done"
- Documentation
- Testing
- Validation

**Comment by melwoodard** [comment]
I think this looks good. @line47 @phiden any thoughts about how this could fold into the pattern lifecycle doc? https://github.com/CMSgov/design-system/wiki/Pattern-lifecycle

File: packages/docs/src/pages/guidelines/component-maturity.md (hunk 1 above)
Comment (line47) [reply packages/docs/src/pages/guidelines/component-maturity.md]: We have documentation outlined here https://github.com/CMSgov/design-system/blob/sw-writing-docs-update/guides/WRITING-DOCUMENTATION.md#documentation-and-guidance

and testing and validation outlined here https://github.com/CMSgov/design-system/blob/sw-propose-new-item-issue-template/.github/ISSUE_TEMPLATE/component_validation_checklist.md

Both are PRs awaiting review

**Comment by line47** [comment]
@mel I think this now folded into the pattern lifecycle doc - which is yet to be published

**Comment by line47** [comment]
I like to get this merged can you take a look @phiden 
@1Copenut I re-requested reviews on this and another PR

**Pull request merged**